bin/pgenv

```diff
@@ -118,15 +135,94 @@ pgenv_check_dependencies(){
     fi
 }
 
+# This function checks for its first argument to be
+# a valid PostgreSQL number, soemthing like 10.5 or 9.5.4.
+# In case no version number is specified, or the
+# version number is not valid (e.g., 10, 0.9) the
+# function aborts the script.
+# If no version at all is specified, the `pgversions` function
+# is invoked to display the user with available versions.
+pgenv_input_version_or_exit() {
+    local version=$1
+
+    if [ -z "$version" ]; then
+        echo "Which version of PostgreSQL do you want?"
+        pgversions
+        exit 1
+    fi
+
+    if [[ $version =~ ^[1-9][0-9]?\.[0-9]+(\.[0-9]+)?$ ]]; then
+        return
+    elif [[ $version =~ ^[1-9][1-9]beta[1-9]?$ ]]; then
+        return
+    else
+        echo "Specified version <$version> does not appear to be a PostgreSQL valid version number"
+        exit 1
+    fi
+}
+
+# This function accepts a version number (that should
+# have been already checked to be valid)
+# and searches for an installation with such
+# a version. If no installation with such version number
+# has been found, the script aborts after
+# showing installed versions
+pgenv_installed_version_or_exit(){
+    local version=$1
+
+    if [ ! -e "pgsql-${version}" ]; then
+        echo "PostgreSQL $version not installed; installed versions:"
+        pgversions
+        exit 1
+    fi
+
+}
+
+# Checks if the user PGENV_PG_USER
+# does exists on this system or aborts the script.
+# This is used to check that the user
+# does exist before a build is started.
+pgenv_existing_user_or_exit(){
+    trap "" ERR
+    id -u $PGENV_PG_USER > /dev/null 2>&1
```

Comment:
What? I don't need the user to exist on the system.

Reply:
Removed by commit b00e958233b3bd27682364e439e990ab110515a9. I agree it could have been to much aggressive, I was "simulating" what other distros does with regard to the need of having a database user that both own the daemon and is the superuser. Since the daemon here is hold by the current user, I agree it could be removed.